### ISSUE TITLE
Chinese support

### ISSUE THREAD

**Issue by dmyers** [cocur/slugify#47]
Is there a way to get Chinese characters to work?

**Comment by danielanteloagra** [comment]
would be interested in this too, how would you go about it?

**Comment by florianeckerstorfer** [comment]
I have little to no knowledge about Chinese, but as far as I understand that Chinese doesn't really have what we call letters. That is, you cannot really transliterate a Chinese symbol into an ASCII character (or a series of characters), because Chinese characters are basically words (or at least part of words). Again I'm not 100% sure.

Take a look at [FaSlugify](https://github.com/ammont/FaSlugify) (which seems to be based on this library for Farsi/Persian. This project includes another project called finglify which translates a Persian/Farsi word into latin characters. It includes a massive [list of words](https://github.com/ammont/finglify/blob/master/resources/words.json). I guess that a Slugify for Chinese would also need such a list.

**Comment by zoransa** [comment]
I have solved it kind of...

    if (!function_exists('slugify')) {

    /**
     * Slugify string for urls
     * @param $str
     * @param string $separator
     * @param string $locale
     * @param bool $lowercase
     * @return string
     */
    function slugify($str, $separator = "-", $locale = 'en', $lowercase = false)
    {
        if (in_array($locale, ['zh'])) {
            $str = transliterator_transliterate('Any-Latin', $str);
        }

        $slugify = new Slugify(null, array('lowercase' => $lowercase));

        $slugify->addRules(
            [
                'א' => '',
                'ב' => 'v',
                'ב' => 'b',
                'ג' => 'g',
                'ג׳' => 'j',
                'ד' => 'd',
                'ד׳' => 'dh',
                'ה' => 'h',
                'ו' => 'v',
                'ז' => 'z',
                'ז׳' => 'zh',
                'ח' => 'h',
                'ט' => 't',
                'י' => 'y',
                'כ' => 'k',
                'ך' => 'kh',
                'ל' => 'l',
                'מ' => 'm',
                'ם' => 'm',
                'נ' => 'n',
                'ן' => 'n',
                'ס' => 's',
                'ע' => '\'',
                'פ' => 'f',
                'ף' => 'f',
                'צ' => 'ts',
                'ץ' => 'ts',
                'ץ׳' => 'tsh',
                'ק' => 'k',
                'ר' => 'r',
                'ש' => 's',
                'ת' => 't'
            ]
        );
        return $slugify->slugify($str, $separator);
    }
    }

This adds rules for Hebrew and in case of Chinese it does transliterate but you have to send 'zh' $locale flag to the function.

Also update your composer and add:

         "require": {
               "ext-intl": "*",

**Comment by gfd6th** [comment]
[Check it on php manul](http://php.net/manual/en/regexp.reference.unicode.php)

```
<?php 
preg_match("/\p{Han}+/u", 'test中文', $result); 
var_dump($result);
```

**Comment by Dimimo** [comment]
Interesting question for I will sooner or later need this. After some research I have found this solution: [http://pastebin.com/zZfci0Dz](url). The URL won't look nice but it is re-translatable from UTF-8 to Chinese without a loss. 

There are easier solutions but they are **not reliable** and even dangerous. Read this first reply to understand why: [http://stackoverflow.com/questions/5998607/conversion-from-simplified-to-traditional-chinese](url). You could run into problems. Something better to avoid as a web-designer. Use numbered url's instead.

**Comment by rny** [comment]
```
		$slugify = new \Cocur\Slugify\Slugify();
		echo $slugify->slugify('中文');
```

echo empty, should be unchanged string '中文'.

**Comment by picks44** [comment]
+1 for the addition of the Chinese/pinyin

**Comment by zoransa** [comment]
In order to make it Chinese/pinyin just modify Chinese string with

$str = transliterator_transliterate('Any-Latin', $str);

don't forget php requires "ext-intl": "*",

**Comment by SuN-80** [comment]
I suppose it could be easy to implement using a class like this as an addon: https://github.com/jifei/Pinyin/blob/master/Pinyin.php

**Comment by SuN-80** [comment]
Added Chinese support (pinyin): #158

**Comment by hifall** [comment]
#158 PR still not approved -- any progress? Thanks!

**Comment by SuN-80** [comment]
I don't know why checks are not successful, but it works: https://getsongbpm.com/artist/zhoujie/3vJn
Unless someone can tell me what's wrong with conflicting files, you'll have to add Chinese rules manually and generate your own set of rules (like i did in the example above) using #158 .

**Comment by florianeckerstorfer** [comment]
Fixed by #195
